torch/ao/quantization/pt2e/qat_utils.py

```diff
@@ -522,11 +522,12 @@ def _fuse_conv_bn_qat(m: GraphModule) -> GraphModule:
     has_bn = any(_is_bn_node(n) for n in m.graph.nodes)
     if not has_bn:
         return m
-    m = _fuse_conv_bn_qat_helper(m, F.conv1d, _conv1d_bn_example_inputs, is_cuda=False)
-    m = _fuse_conv_bn_qat_helper(m, F.conv2d, _conv2d_bn_example_inputs, is_cuda=False)
-    if torch.cuda.is_available():
-        m = _fuse_conv_bn_qat_helper(m, F.conv1d, _conv1d_bn_example_inputs, is_cuda=True)
-        m = _fuse_conv_bn_qat_helper(m, F.conv2d, _conv2d_bn_example_inputs, is_cuda=True)
+    is_cuda_options = [True, False] if torch.cuda.is_available() else [False]
```

Comment:
one question here is do we want to separate model preparation, v.s. model execution? right now it seems that how the model is prepared depends on the status of whether cuda is available or not when we run the model preparation code, which doesn't make a lot of sense to me. I'm wondering whether we need to make this decision more explicit for users here.

Reply:
This is currently needed because we have different batch norm ops depending on cuda vs cpu, so we need to match both versions. After batch norm consolidation we might not need this anymore. However, I'm hesitant to make the general assumption that batch norm is the only op we need to worry about. In general, export gives us a graph specialized to the device, and I feel there's not much we can do about that.

Reply:
so my point is model preparation/quantization should be decoupled from the whether the model happens to be prepared/quantized in cuda or cpu right?

Reply:
I agree at a high level. The way we get the cuda version here is by calling `example_inputs.cuda()` first before exporting. I don't think that will work on a CPU-only machine. Also this is just refactoring here so if we decide to change this let's do it in a separate PR

Reply:
oh I see, sounds good